File: gen_pdf.py
from fpdf import FPDF
import re

pdf = FPDF()
pdf.add_page()
pdf.set_auto_page_break(auto=True, margin=15)

pdf.set_font("Helvetica", size=12)

with open("complex_test_contract.md", "r", encoding="utf-8") as f:
    for line in f:
        line = line.strip()
        if not line:
            pdf.ln(5)
            continue
            
        if line.startswith("# "):
            pdf.set_font("Helvetica", style="B", size=16)
            pdf.multi_cell(0, 10, line[2:])
            pdf.set_font("Helvetica", size=12)
        elif line.startswith("## "):
            pdf.ln(5)
            pdf.set_font("Helvetica", style="B", size=14)
            pdf.multi_cell(0, 10, line[3:])
            pdf.set_font("Helvetica", size=12)
        else:
            # Remove bold/italic markdown characters for simple pdf rendering
            clean_line = re.sub(r'[*_]{1,2}', '', line)
            pdf.multi_cell(0, 8, clean_line)

pdf.output("complex_test_contract.pdf")
print("PDF created successfully!")

File: gen_pdf.js
const PDFDocument = require('pdfkit');
const fs = require('fs');

const doc = new PDFDocument({ margin: 50 });
doc.pipe(fs.createWriteStream('complex_test_contract.pdf'));

const text = fs.readFileSync('complex_test_contract.md', 'utf8');

doc.font('Helvetica-Bold').fontSize(16).text('MASTER SERVICES AND DATA PROCESSING AGREEMENT', { align: 'center' });
doc.moveDown(2);

const lines = text.split('\n');

for (const line of lines) {
    if (line.startsWith('# ')) continue; // Skip title as we did it above
    
    if (line.startsWith('## ')) {
        doc.moveDown(1);
        doc.font('Helvetica-Bold').fontSize(14).text(line.replace('## ', ''));
        doc.moveDown(0.5);
    } else if (line.trim().length > 0) {
        let cleanLine = line.replace(/\*\*/g, '').replace(/\*/g, '');
        doc.font('Helvetica').fontSize(11).text(cleanLine, {
            align: 'justify',
            lineGap: 4
        });
        doc.moveDown(0.5);
    }
}

doc.end();
console.log("PDF created successfully using PDFKit!");
